### PR TITLE
Update : sdk bump, ibc version and band oracle changes,tendermint upg…

### PR DESCRIPTION
**Pull request by dheerajkd30**
PR includes 

1. Sdk bump to 45.11
2. ibc version upgrade to v4.2.0
3. wasm version upgrade
4. tendermint version upgrade
5. refactor and remove unused code

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -795,7 +795,7 @@ func New(
 
 	ibcRouter.AddRoute(ibctransfertypes.ModuleName, transferIBCModule)
 	ibcRouter.AddRoute(bandoraclemoduletypes.ModuleName, bandOracleIBCModule)
-	ibcRouter.AddRoute(wasm.ModuleName, wasm.NewIBCHandler(app.WasmKeeper, app.IbcKeeper.ChannelKeeper))
+	ibcRouter.AddRoute(wasm.ModuleName, wasm.NewIBCHandler(app.WasmKeeper, app.IbcKeeper.ChannelKeeper, app.IbcKeeper.ChannelKeeper))
```
Comment (vishnukumavat) [inline app/app.go:798]: `app.IbcKeeper.ChannelKeeper` 2 times? is the parameter same for `appVersionGetter`

File: x/bandoracle/types/errors.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 var ErrorUnknownProposalType = errors.Register(ModuleName, 401, "unknown proposal type")
 
 var (
-	ErrSample         = errors.Register(ModuleName, 1100, "sample error")
-	ErrInvalidVersion = errors.Register(ModuleName, 1501, "invalid version")
+	ErrSample             = errors.Register(ModuleName, 1100, "sample error")
```
Comment (vishnukumavat) [inline x/bandoracle/types/errors.go]: @pratikasr please replace this error with relevant message wherever it is being used

**Review by vishnukumavat — COMMENTED** [review]
Please check commments

File: app/app.go (hunk 1 above)
Comment (dheerajkd30) [reply app/app.go:798]: This is correct implementation and verified . We can change it in next release if we plan to implement

File: x/bandoracle/types/errors.go (hunk 2 above)
Comment (dheerajkd30) [reply x/bandoracle/types/errors.go]: Done.

**Review by vishnukumavat — APPROVED** [review]
LGTM

**Review by cgsingh33 — APPROVED** [review]

**Review by pratikasr — APPROVED** [review]

**Pull request merged**